From c08f4e55737a3e905b5a30f9eb72d4f8beaa292e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Tue, 4 Dec 2012 17:30:17 +0100
Subject: [PATCH] Do not wrap number of bytes to copy silently

32-bit x86 architecture has IV_MAX 2 less than I32_MAX. This allows to
initialize less bytes than requested (repeat counter 2^32-1..2^32+1 is
affected). This patch changes the silent wrap to noisy croak.

See <https://bugzilla.redhat.com/show_bug.cgi?id=720644>.
---
 pp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pp.c b/pp.c
index 28bbfe8..460e410 100644
--- a/pp.c
+++ b/pp.c
@@ -1490,7 +1490,7 @@ PP(pp_repeat)
 	 if (SvUOK(sv)) {
 	      const UV uv = SvUV(sv);
 	      if (uv > IV_MAX)
-		   count = IV_MAX; /* The best we can do? */
+		   Perl_croak_nocontext("%s", PL_memory_wrap);
 	      else
 		   count = uv;
 	 } else {
-- 
1.7.11.7

